autopush/websocket.py

```diff
@@ -270,21 +270,26 @@ def process_hello(self, data):
         self.uaid = uaid
 
         connect = data.get("connect")
-        if connect and self.ap_settings.pinger:
+        if connect and self.ap_settings.bridge:
             self.transport.pauseProducing()
-            d = deferToThread(self.ap_settings.pinger.register, uaid, connect)
+            d = deferToThread(self.ap_settings.bridge.register, uaid, connect)
             d.addCallback(self._check_router, True)
             d.addErrback(self.err_hello)
         else:
             self._check_router(False)
 
-    def _check_router(self, paused=False):
+    def _check_router(self, paused=False, bridge_register=None):
         if paused:
             self.transport.resumeProducing()
+        if bridge_register is not None and bridge_register is False:
```

Comment:
I've argued about this with myself a fair bit. Push can fail back to maintaining it's own socket if bridge registration fails. It's not as efficient, but it's possible. Bridges are a bit of a cost savings/efficiency thing both for the client and for the server. That said, it's certainly possible for the bridge to be broken for any number of reasons (registration failure, provider termination, etc.) All of which should probably be factored somehow.

I think that registration failure needs to be communicated back to the client, but that's also why I don't terminate the socket connection on a bridge failure. The client is certainly free to, at that point. 

The way I view it, the client is really the source of truth about how critical a given bridge really is. Granted, this is purely a subjective ruling so I'm happy to change things if there's good logic otherwise.